tests/test_diffraction_StaticStructureFactorDebye.py

```diff
@@ -0,0 +1,215 @@
+import pytest
+
+import freud
+
+matplotlib.use("agg")
+
+
+def _validate_debye_method(system, bins, k_max, k_min):
+    """Validation of the static structure calculation.
+
+    This method is a pure Python reference implementation of the Debye method
+    implemented in C++ in freud.
+
+    Args:
+        system:
+            Any object that is a valid argument to
+            :class:`freud.locality.NeighborQuery.from_system`.
+        bins (unsigned int):
+            Number of bins in :math:`k` space.
+        k_max (float):
+            Maximum :math:`k` value to include in the calculation.
+        k_min (float):
+            Minimum :math:`k` value to include in the calculation.
+    """
+    system = freud.locality.NeighborQuery.from_system(system)
+    N = len(system.points)
+
+    Q = np.linspace(k_min, k_max, bins, endpoint=False)
+    Q += (k_max - k_min) / bins / 2
+    S = np.zeros_like(Q)
+
+    # Compute all pairwise distances
+    distances = system.box.compute_all_distances(system.points, system.points).flatten()
+
+    for i, q in enumerate(Q):
+        S[i] += np.sum(np.sinc(q * distances / np.pi)) / N
+
+    return Q, S
+
+
+class TestStaticStructureFactorDebye:
+    def test_compute(self):
+        sf = freud.diffraction.StaticStructureFactorDebye(1000, 100)
+        box, positions = freud.data.UnitCell.fcc().generate_system(4)
+        sf.compute((box, positions))
+
+    def test_debye_validation(self):
+        """Validate the Debye method against a Python implementation."""
+        bins = 1000
+        k_max = 100
+        k_min = 0
+        sf = freud.diffraction.StaticStructureFactorDebye(bins, k_max, k_min)
+        box, points = freud.data.UnitCell.fcc().generate_system(4, sigma_noise=0.01)
+        system = freud.locality.NeighborQuery.from_system((box, points))
+        sf.compute(system)
+        Q, S = _validate_debye_method(system, bins, k_max, k_min)
+        npt.assert_allclose(sf.bin_centers, Q)
+        npt.assert_allclose(sf.S_k, S, rtol=1e-5, atol=1e-5)
+
+    def test_debye_ase(self):
+        """Validate Debye method agains ASE implementation"""
+        bins = 1000
+        k_max = 100
+        k_min = 0
+        box, points = freud.data.UnitCell.fcc().generate_system(4, sigma_noise=0.01)
+        box.periodic = False
+        print(box)
+        system = freud.locality.NeighborQuery.from_system((box, points))
+        sf = freud.diffraction.StaticStructureFactorDebye(bins, k_max, k_min)
+        sf.compute(system)
+        import ase
+        from ase.utils.xrdebye import XrDebye
+
+        atoms = ase.Atoms(
+            positions=points, pbc=True, cell=box.L, numbers=np.ones(len(points))
+        )
+        xrd = XrDebye(atoms=atoms, wavelength=1.0, method="??", damping=0)
+        S_ase = []
+        for i in sf.bin_centers / (2 * np.pi):
+            S_ase.append(xrd.get(i))
+        S_ase = np.asarray(S_ase) / len(points)
+        npt.assert_allclose(sf.S_k, S_ase, rtol=1e-5, atol=1e-5)
```

Comment:
Make this test have a soft dependency on ase if possible. See examples:
- Optional module: https://github.com/glotzerlab/freud/blob/ea8c460ddc81b153616d2695fa68627d653a49b3/tests/integration/test_reader_integrations.py#L34
- Optional function/class: https://github.com/glotzerlab/freud/blob/ea8c460ddc81b153616d2695fa68627d653a49b3/tests/test_environment_LocalDescriptors.py#L336


```suggestion
    def test_debye_ase(self):
        """Validate Debye method agains ASE implementation"""
        ase = pytest.importorskip("ase")
        xrdebye = pytest.importorskip("ase.utils.xrdebye")
        bins = 1000
        k_max = 100
        k_min = 0
        box, points = freud.data.UnitCell.fcc().generate_system(4, sigma_noise=0.01)
        box.periodic = False
        print(box)
        system = freud.locality.NeighborQuery.from_system((box, points))
        sf = freud.diffraction.StaticStructureFactorDebye(bins, k_max, k_min)
        sf.compute(system)
        atoms = ase.Atoms(
            positions=points, pbc=True, cell=box.L, numbers=np.ones(len(points))
        )
        xrd = xrdebye.XrDebye(atoms=atoms, wavelength=1.0, method="??", damping=0)
        S_ase = []
        for i in sf.bin_centers / (2 * np.pi):
            S_ase.append(xrd.get(i))
        S_ase = np.asarray(S_ase) / len(points)
        npt.assert_allclose(sf.S_k, S_ase, rtol=1e-5, atol=1e-5)
```

Reply:
I think this is generally on the right track with most parameters should set to 1 or 0, e.g. wavelength 1, damping 0, method `None` (not the default of Iwasa). I would also try `alpha=0` so that there is no angular damping of scattering intensity?

You can probably use `calc_pattern` with SAXS and provide an array of `q` values instead of repeatedly calling `get(x)`. SAXS lets you specify the values of `q` directly. The XRD mode requires you to specify the angles, which is not what we want ([see description here](https://wiki.fysik.dtu.dk/ase/ase/xrdebye.html#ase.utils.xrdebye.XrDebye.calc_pattern)).

Reply:
Agreed on parameters. I updated `method` to `None` and added `alpha=0` . The reason I opted out from using `calc_pattern` was because that function does division of q by 2 \pi under the hood which might made it hard to track 2 \pi 's. It also uses `get(x)` under the hood anyways. I updated the test so that it uses `calc_pattern` now with some extra comments. It does look cleaner though IMO.